File: streaming-overlay-frontend/src/types/cricket.ts
export interface Player {
  id: string;
  name: string;
  runs?: number;
  balls?: number;
  fours?: number;
  sixes?: number;
  strikeRate?: number;
  isStriker?: boolean;
}

export interface Bowler {
  id: string;
  name: string;
  overs: number;
  maidens: number;
  runs: number;
  wickets: number;
  economy: number;
}

export interface Team {
  id: string;
  name: string;
  score: number;
  wickets: number;
  overs: number;
  runRate: number;
  players: Player[];
  bowlers: Bowler[];
}

export interface CricketMatch {
  id: string;
  team1: Team;
  team2: Team;
  currentInnings: 1 | 2;
  matchType: "T20" | "ODI" | "TEST";
  oversLimit?: number;
  target?: number;
  requiredRunRate?: number;
  lastBall?: string; // For ball-by-ball commentary
  currentOver: string[]; // Array of last 6 balls
  partnership: {
    runs: number;
    balls: number;
  };
}

File: streaming-overlay-backend/src/streaming_overlay/models/cricket.py
from typing import Literal, Optional

from pydantic import BaseModel


class Player(BaseModel):
    id: str
    name: str
    runs: Optional[int] = 0
    balls: Optional[int] = 0
    fours: Optional[int] = 0
    sixes: Optional[int] = 0
    strike_rate: Optional[float] = 0.0
    is_striker: Optional[bool] = False


class Bowler(BaseModel):
    id: str
    name: str
    overs: float = 0.0
    maidens: int = 0
    runs: int = 0
    wickets: int = 0
    economy: float = 0.0


class Team(BaseModel):
    id: str
    name: str
    score: int = 0
    wickets: int = 0
    overs: float = 0.0
    run_rate: float = 0.0
    players: list[Player]
    bowlers: list[Bowler]


class CricketMatch(BaseModel):
    id: str
    team1: Team
    team2: Team
    current_innings: Literal[1, 2] = 1
    match_type: Literal["T20", "ODI", "TEST"]
    overs_limit: Optional[int]
    target: Optional[int]
    required_run_rate: Optional[float]
    last_ball: Optional[str]
    current_over: list[str] = []
    partnership: dict = {"runs": 0, "balls": 0}
